vmdb/app/controllers/api_controller/parser.rb

```diff
@@ -107,6 +107,17 @@ def href_collection_id(path)
       subcollection ? [subcollection.to_sym, s_id] : [collection.to_sym, c_id]
     end
 
+    def parse_id(resource, collection)
+      return nil if resource.blank?
+
+      href = resource["href"]
+      if href.present?
+        return href.match(%r{^.*/#{collection}/[0-9]+$}) ? href.split('/').last : nil
+      end
+
+      resource["id"].present? ? resource["id"] : nil
```

Comment:
good point, now that the spec clearly separates id/href, If specified, we can take is only if kind_of?(Integer), otherwise nil.